engine/class_modules/monk/sc_monk.cpp

```diff
@@ -327,16 +344,159 @@ struct monk_action_t : public Base
+      {
+        if ( td->debuff.bonedust_brew->check() )
+          bdb_targets++;
+        if ( td->debuff.mark_of_the_crane->check() )
+          MotC_stacks++;
+      }
+    }
+
+    if ( bdb_targets == 0 )
+      return 0;
+
+    double H = 1 / p()->composite_melee_haste();
+    double M = 1 + p()->composite_mastery_value();
+
+    // Delay when chaining SCK (approximately 200ms from in game testing)
+    double eps = 0.2;
+
+    auto tiger_palm_AP_ratio_by_aura = p()->spec.tiger_palm->effectN( 1 ).ap_coeff() *
+                                       ( 1 + p()->spec.windwalker_monk->effectN( 1 ).percent() ) *
+                                       ( 1 + p()->spec.windwalker_monk->effectN( 5 ).percent() ) *
+                                       ( 1 + p()->spec.windwalker_monk->effectN( 16 ).percent() ) *
+                                       ( 1 + p()->spec.windwalker_monk->effectN( 18 ).percent() );
+    auto SCK_AP_ratio_by_aura = p()->spec.spinning_crane_kick->effectN( 2 ).ap_coeff() *
+                                // 4 ticks
+                                4 * ( 1 + p()->spec.windwalker_monk->effectN( 2 ).percent() ) *
+                                ( 1 + p()->spec.windwalker_monk->effectN( 8 ).percent() );
+
+    auto CS_bonus  = p()->conduit.calculated_strikes->ok() ? p()->conduit.calculated_strikes.percent()
+                                                           : p()->passives.cyclone_strikes->effectN( 1 ).percent();
+    auto BMH_bonus = p()->conduit.bone_marrow_hops->ok() ? p()->conduit.bone_marrow_hops.percent() : 0;
+    auto CO_bonus  = p()->conduit.coordinated_offensive->ok() ? p()->conduit.coordinated_offensive.percent() : 0;
+
+    // sqrt scaling
+    auto N_effective_targets_above = 5 * pow( ( N / 5 ), 0.5 );
+    auto N_effective_targets       = N < N_effective_targets_above ? N : N_effective_targets_above;
+
+    auto MotC_bonus_per_target = .18 + CS_bonus;
+    auto MotC_bonus            = MotC_bonus_per_target * MotC_stacks;
+    auto MotC_multiplier       = 1 + MotC_bonus;
+
+    auto SCK_dmg_total = ( N_effective_targets * SCK_AP_ratio_by_aura * MotC_multiplier ) -
+                         ( 1.1 / 1.676 * .81 / 2.5 * 1.5 );  // Subtract auto attacks
+    auto tp_over_sck = tiger_palm_AP_ratio_by_aura / SCK_dmg_total;
+
+    auto flAmp = 1 + p()->composite_damage_versatility();
+
+    if ( p()->buff.invoke_xuen->check() )
+    {
+      // Tiger Lightning
+      flAmp *= ( 1 + p()->spec.invoke_xuen_2->effectN( 2 ).percent() );
+    }
+
+    if ( p()->buff.storm_earth_and_fire->check() )
+    {
+      auto sef_multiplier = ( 1 + p()->spec.storm_earth_and_fire->effectN( 1 ).percent() ) * 3;
+      flAmp *= sef_multiplier;
+      flAmp *= ( 2 * CO_bonus + 1 ) / 3;
+    }
+
+    flAmp = flAmp * ( 1 + ( .5 * .4 * ( static_cast<double>( bdb_targets ) / N ) * ( 1 + BMH_bonus ) ) );
+
+    auto energy_regen_per_second = p()->resource_regen_per_second( RESOURCE_ENERGY );
+
+    // TP->SCK
```

Comment:
It would be best if you can move the following 3 rdps calculations into some shared function that you call to get the rdps values. Should make your code much smaller and easier to understand.